noir-projects/noir-contracts/contracts/avm_test_contract/src/main.nr

```diff
@@ -210,10 +216,13 @@ pub contract AvmTest {
     }
 
     // Helper functions to demonstrate an internal call stack in error messages
+    #[contract_library_method]
     fn inner_helper_with_failed_assertion() {
         let not_true = false;
         assert(not_true == true, "This assertion should fail!");
     }
+
+    #[contract_library_method]
```

Comment:
@fcarreiro these functions were being called directly from public functions like any regular noir function, but any fn directly inside the `contract` mod gets made a contract entrypoint automatically (weird I know). We now detect and warn when we see these, so I added the `contract_library_method` attribute, which makes the fn _not_ be an entrypoint (weird I know), which I assume is what you originally intended.

Reply:
Thanks yes! I guess they should be unconstrained as well? In any case if this works we can check later

Reply:
No need, they'll be converted to unconstrained automatically if called from an unconstrained fn (like public fns are). We'd only need to make this helper unconstrained if we needed to use any of the unconstrained features (e.g. `while` or `break`).